test/zcl_abapgit_integration_git.clas.testclasses.abap

```diff
@@ -10,13 +25,27 @@ CLASS ltcl_test IMPLEMENTATION.
 
   METHOD test01.
 
-    DATA lv_bar TYPE i.
+    CONSTANTS lc_url TYPE string VALUE 'https://github.com/abapGit/abapGit.git'.
+    DATA lv_ref TYPE string.
+    DATA li_memory TYPE REF TO lcl_memory_settings.
+
+    CREATE OBJECT li_memory.
+    zcl_abapgit_persist_injector=>set_settings( li_memory ).
+
+    " abc lv_ref = zcl_abapgit_git_transport=>branches( lc_url )->get_head_symref( ).
```

Comment:
shouldn't be a test here?
And what about commented code?

Reply:
that's the part I'm working on :)

its in /test/ so its not installed into customer systems

it should probably have been in a separate PR, but consider it as a playground for now

https://github.com/abapGit/abapGit/tree/hvam/upd3007/test#integration-testing---git-protocol

Reply:
after getting the changes from `main`, the first test is running 😄